src/components/reusable/WeatherIcon/WeatherIcon.jsx

```diff
@@ -0,0 +1,40 @@
+import React from 'react';
+import { ICON_HEIGHT_BIG, ICON_WIDTH_BIG, ICON_HEIGHT_SMALL, ICON_WIDTH_SMALL } from '../../../constants';
+import { weatherIcon } from './WeatherIcon.scss';
+
+class WeatherIcon extends React.Component {
+    weatherIcons = null;
+
+    constructor(props) {
+        super(props);
+        const iconInfo = this.props.iconInfo;
+        const isCurrentCity = this.props.isCurrentCity;
+        this.weather = iconInfo.weather;
+        this.iconType = iconInfo.iconType;
+        this.width = isCurrentCity ? ICON_WIDTH_BIG : ICON_WIDTH_SMALL;
```

Comment:
Can you elaborate why you don;t use prop values directly but instead store them on instance in construcotr?